### PR TITLE
Use new ad request format by default in TheMediaGrid Bid Adapter

### PR DESCRIPTION
**Pull request by TheMediaGrid**
## Type of change

- [x] Bugfix
- [x] Refactoring (no functional changes, no api changes)
- [x] Feature

## Description of change

Use new ad request format by default in TheMediaGrid Bid Adapter.
Fix logic with timeout.
Update userId format in the ad request
Added bidFloor parameter
- test parameters for validating bids
```
{
    bidder: "grid",
    params: {
        uid: '1',
        bidFloor: 0.5
    }
}
```

- grid-tech@themediagrid.com

- [x] official adapter submission

- A link to a PR on the docs repo - https://github.com/prebid/prebid.github.io/pull/2413

### CODE REVIEW

File: modules/gridBidAdapter.js [hunk 1]
```diff
@@ -156,18 +156,68 @@ export const spec = {
...
+        userExt.eids.push({
+          source: 'adserver.org', // Unified ID
+          uids: [{
+            id: userId.tdid,
+            ext: {
+              rtiPartner: 'TDID'
+            }
+          }]
+        });
+        // userExt.unifiedid = userId.tdid;
```
Comment (ncolletti) [inline modules/gridBidAdapter.js:174]: Please remove comments, there are a few in this file in the new user id code

Comment (ncolletti) [inline modules/gridBidAdapter.js:174]: Is the commented code necessary on lines 171, 183, 194, 220? Looks like it's left over from dev?

File: modules/gridBidAdapter.js [hunk 2]
```diff
@@ -44,19 +43,237 @@ export const spec = {
...
+      let impObj = {
+        id: bidId,
+        tagid: uid.toString(),
+        ext: {
+          divid: adUnitCode
+        }
+      };
+
+      if (bidFloor) {
+        impObj.bidfloor = bidFloor;
```
Comment (ncolletti) [inline modules/gridBidAdapter.js:97]: Please update to support the [Price Floor module ](https://docs.prebid.org/dev-docs/modules/floors.html)as per other bidders and default to the bidFloor param value if no Floor value is passed in the module. You can refer to other bidders for example of how to implement.

File: modules/gridBidAdapter.js (hunk 1 above)
Comment (TheMediaGrid) [reply modules/gridBidAdapter.js]: Comments have been removed

File: modules/gridBidAdapter.js (hunk 2 above)
Comment (TheMediaGrid) [reply modules/gridBidAdapter.js]: Fixed

**Comment by ncolletti** [comment]
Thanks this looks good.. please resolve the conflicts in the PR and I will merge in.

**Comment by TheMediaGrid** [comment]
Conflict resolved

**Pull request merged**